### ISSUE TITLE
[WRONG_INDENTATION] continuation indent expected in if expression

### ISSUE THREAD

**Issue by icemachined** [saveourtool/diktat#1351]
## Expected behavior
there should be no error after idea style applying

## Observed behavior
  "level": "error",
          "locations": [
            {
              "physicalLocation": {
                "artifactLocation": {
                  "uri": "diktat-rules\\src\\main\\kotlin\\org\\cqfn\\diktat\\ruleset\\rules\\chapter2\\kdoc\\KdocComments.kt",
                  "uriBaseId": "%SRCROOT%"
                },
                "region": {
                  "startColumn": 1,
                  "startLine": 116
                }
              }
            }
          ],
          "message": {
            "text": "[WRONG_INDENTATION] only spaces are allowed for indentation and each indentation should equal to 4 spaces (tabs are not allowed): expected 12 but was 16"
          },
          "ruleId": "diktat-ruleset:zct-indentation"
        
## Steps to Reproduce
If you apply idea formatting in KdocComments you have the following piece of code
```
        if (valueParameterNode.parents().none { it.elementType == PRIMARY_CONSTRUCTOR } ||
            !valueParameterNode.hasChildOfType(VAL_KEYWORD) && 
            !valueParameterNode.hasChildOfType(VAR_KEYWORD)
        ) {
            return
        }
```
but diktat report error as far as it expect continuation indent in the last line of if expression like this
```
        if (valueParameterNode.parents().none { it.elementType == PRIMARY_CONSTRUCTOR } ||
            !valueParameterNode.hasChildOfType(VAL_KEYWORD) && 
                !valueParameterNode.hasChildOfType(VAR_KEYWORD)
        ) {
            return
        }
```

## Environment information
* diktat version: 1.1.1-SNAPSHOT
* build tool (maven/gradle):
* how is diktat run (CLI, plugin, etc.):
* kotlin version:
* operating system:
* link to a project (if your project is public):

**Comment by 0x6675636b796f75676974687562** [comment]
No. 1 is clearly a duplicate of #1336.

**Comment by 0x6675636b796f75676974687562** [comment]
**No. 1** is a separate matter. While all these are correctly formatted (_IDEA_ and _diKTat_ behave consistently):

```kotlin
if (true &&
    true &&
    false) {
    return
}

if (true ||
    true ||
    false) {
    return
}

if (true &&
    true ||
    false) {
    return
}
```

&mdash; here's the minimal repro:

```kotlin
if (true ||
    true &&
        false) {
    return
}
```

Apparently, this is something related to operator priorities, since adding parentheses around the 1st two booleans immediately "fixes" the issue (and modifies the logic):

```kotlin
if ((true ||
    true) &&
    false) {
    return
}
```
